RIOT-OS/RIOT
# drivers: stm32 eth peripheral driver

### Contribution description
Hi All,

as discussed in #5643 this is a new PR based on @lebrush's work.
The original PR implemented an ethernet peripheral driver for stm32f4 boards. 

Changes:

1. Rebase to current master branch
1.1. Rework driver to the new iolist netdev interface
1.2. Rework driver to use new vendor headers
2. Use dynamic MAC Addresses based on board uid
3. Extend driver to work with stm32f7 boards as well
4. Add GNRC Testcase

Open Tasks:
1. Adapt to DMA API?
2. Fix static buffer used as workaround for iolist
3. ~~Maybe use network interface as default interface for boards with on board eth?~~ Implemented
4. Fix a few details (mentioned in in-line comments)


### Testing procedure
1. Go to tests/stm32_eth_lwip or stm32_eth_gnrc, build an flash.
2. Use ICMP or UDP on port 12345 to test the echo server.

I testes with an Nucleo-F767ZI. Original work was tested with STM32F4Discovery. IHMO This should work with every F4 and F7 Board with ethernet peripheral. Would be great if somebody with access to any F4 or F7 hardware could test as well.

### Issues/PRs references
see #5643 
All Credits to the great work of @lebrush who saved my thesis.
Also involved: @MrKevinWeiss  @aabadie @smlng
